ec/ecresource/projectresource/create_test.go

```diff
@@ -0,0 +1,302 @@
+package projectresource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal/gen/serverless/resource_elasticsearch_project"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestCreate(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	type testData struct {
+		api           api[resource_elasticsearch_project.ElasticsearchProjectModel]
+		modelHandler  modelHandler[resource_elasticsearch_project.ElasticsearchProjectModel]
+		req           resource.CreateRequest
```

Comment:
IMO `req` can be removed (its value isn't used) with the EXPECT call changed to something like `handler.EXPECT().ReadFrom(ctx, tfsdk.Plan{}).Return(nil, readDiags)` (the actual return values depend on the test case)

Reply:
I see what you mean. As the code is, the test makes an assertion that "Create calls ReadFrom with the Plan from the `req` parameter". For example, this test would fail if Create altered the Plan in some way prior to calling ReadFrom. 

I'm happy to make the change you're suggesting, this form of assertion has the tendency to be brittle, but the specific value is currently being used, and is providing some additional value to the test. WDYT?

Reply:
Let's keep it as is.